### DISCUSSION TITLE
How do I get the group ID?

### DISCUSSION THREAD

**Discussion by emmenlau** [AsamK/signal-cli#1196] [Q&A] [answered]
Sorry, I'm a real beginner with signal. I have an account already on phone and desktop, and this account is member of a number of groups. I registered the same account with signal-cli. How would I send messages to the groups? I'm lacking the group ID.

When trying to list the known groups, nothing is returned:
```
#> signal-cli --account "+49123456789" listGroups --detailed
```

The same with the jsonRpc interface:
```
#> echo '{"jsonrpc":"2.0","method":"listGroups"}' | signal-cli --account "+49123456789" jsonRpc
```

The documentation mentions that the group ID is base64-encoded, but if I try that the naiive way from the group name, it does not work either:
```
#> signal-cli --account "+49123456789" listGroups --group-id=$(echo "My Group Name"|base64) --detailed
```

**Comment by exquo** [comment, marked as answer]
> I registered the same account with signal-cli

Running `signal-cli register ...` would un-register that number on the smartphone: only one device can "register" a number; the other devices have to "link" with it.

If you have actually run `signal-cli link ...` and don't get anything in `listGrougs` or `listContacts`, then run `signal-cli receive` once after linking.

**Reply by emmenlau** [reply]
> > I registered the same account with signal-cli
> 
> Running `signal-cli register ...` would un-register that number on the smartphone: only one device can "register" a number; the other devices have to "link" with it.

Aha, that was very helpful! I was indeed seeing that my phone was unregistered from the number but did not know why that would happen. Now everything makes sense, thanks!

**Reply by ahsanali17** [reply]
This helped remove some confusion for me too. It is not very clear in the documentation that we should not register again if we are using 1 mobile number to link two or more devices.
